README.md

```diff
@@ -2,31 +2,428 @@
+   the breed, so the second constructor defaults the `breed` field to "unknown".
+
+
+
+## Comprehension Check
+
+<details>
+    <summary>What is printed when the <code>main</code> method executes?</summary>
+
+  <p>Whiskers,true,squeaky mouse</p>
+  <p>Constructor #1 is executed since only two arguments are passed into the method.</p>
+
+</details>
+
+<br>
+
+```java
+public class Cat {
+    private String name;
+    private boolean purring;
+    private String favoriteToy;
+
+    // Constructor #1
+    public Cat(String name, boolean purring) {
+        this.name = name;
+        this.purring = purring;
+        this.favoriteToy = "squeaky mouse";
+    }
+
+    // Constructor #2
+    public Cat(String name, boolean purring, String favoriteToy) {
+        this.name = name;
+        this.purring = purring;
+        this.favoriteToy = favoriteToy;
+    }
+
+    public static void main(String[] args) {
+        Cat kitty = new Cat("Whiskers", true);
+        System.out.println(kitty.name + "," + kitty.purring + "," + kitty.favoriteToy);
+    }
+}
+```
+
+## Constructor Chaining
+
+We have some code duplication between the two constructors, specifically the
+two statements that assign `name` and `age`.  We can do something
+called *constructor chaining*, where one constructor calls another to initialize
+the variables.
+
+We can have the constructor that takes two parameters (Constructor #2) call the
+constructor that takes four parameters (Constructor #1).  This must be done as
+the *first line of code* in the constructor body and the method name `this`
+is used to call the constructor, i.e.  `this(name, "unknown", age, false);`.
+
+```java
+// Constructor #1
+public Dog(String name, String breed, int age, boolean waggingTail) {
+    this.name = name;
+    this.breed = breed;
+    this.age = age;
+    this.waggingTail = waggingTail;
+}
+
+// Constructor #2
+public Dog(String name, int age) {
+    this(name, "unknown", age, false);  // Call constructor #1
+}
+```
+
+## Default Constructor Revisited
+
+If we define a constructor that has one or more parameters,
+the compiler does not generate a default no-args constructor.
+Thus, we will get a compile-time error if we try to call the no-args constructor `new Dog()`:
+
+```java
+public static void main(String[] args) {
+    Dog bigDog = new Dog("Fido", "Great Dane", 8, true);
+    Dog smallDog = new Dog("Sweetie", "Miniature Poodle", 2, false);
+    Dog mediumDog = new Dog("Honey", 10);
+    Dog xlargeDog = new Dog();  //ERROR: no default constructor
+}
+```
+
+We must explicitly add a no-args constructor  to the `Dog` class
+to avoid the error.
+
+
+## Code Check
+
+Our final version of the `Dog` class is:
+
+```java
+public class Dog {
+
+    private String name;
+    private String breed;
+    private int age;
+    private boolean waggingTail;
+
+    // Constructor #1
+    public Dog(String name, String breed, int age, boolean waggingTail) {
+        this(name, age); //calls Constructor #2
```

Comment:
Should the final version of the `Dog` class have constructor 2 call constructor 1?

Reply:
Yes, thanks!